Sources/Basic/FileSystem.swift

```diff
@@ -101,6 +101,9 @@ public protocol FileSystem {
     
     /// Check whether the given path is accessible and a file.
     func isFile(_ path: AbsolutePath) -> Bool
+    
+    /// Check to determine whether the given path is an accessible and executable file
+    func isExecutableFile(_ path: AbsolutePath) -> Bool
```

Comment:
migrated the Filesystem protocol extension, implementation and related tests to another PR: https://github.com/apple/swift-package-manager/pull/997